src/main/java/training/bank/Bank.java

```diff
@@ -0,0 +1,61 @@
+package training.bank;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.security.InvalidParameterException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class Bank {
+    private static final Logger LOGGER = LogManager.getLogger();
+    private HashMap<String, Account> accounts = new HashMap<>(); // Is this better than HashSet?
```

Comment:
The difference between `HashMap` and `HashSet` is that it allows you to associate each item of the set with a key that can use to look it up quickly - and since we want to look up accounts by name, `HashMap` is a good fit.